org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/MavenPluginActivator.java

```diff
@@ -358,27 +191,10 @@ public void stop(BundleContext context) throws Exception {
     }
     context.removeBundleListener(bundleListener);
```

Comment:
```suggestion
    trackers.values().forEach(ServiceTracker::close);
    trackers.clear();
```
The service-trackers should be closed and the class objects be release here, shouldn't they?
The latter is probably not that important because I suspect the Activator instance will be eligible for GC soon after the bundle was stopped.

Reply:
All services are automatically disposed after the bundle is stopped and all OSGI frees all references to classloaders and objects so they become eligible for garbage collection.